File: Java/Largest-Number.java
public class Solution {
    /**
     *@param num: A list of non negative integers
     *@return: A string
     */
    private class StringComparator implements Comparator<String> {
        public int compare(String s1, String s2) {
            return (s2 + s1).compareTo(s1 + s2);
        }
    }
    public String largestNumber(int[] num) {
        // write your code here
        ArrayList<String> res = new ArrayList<String>();
        for (int i = 0; i < num.length; i ++) {
            res.add(String.valueOf(num[i]));
        }
        Collections.sort(res, new StringComparator());
        StringBuffer str = new StringBuffer();
        for (int i = 0; i < res.size(); i ++) {
            str.append(res.get(i));
        }
        String result = str.toString();
        int index = 0;
        while (index < result.length() && result.charAt(index) == '0') {
            index++;
        }
        if (index == result.length()) {
            return "0";
        }
        return result.substring(index);
    }
}

File: Cpp/Largest-Number.cpp
class Solution {
public:
    /**
     *@param num: A list of non negative integers
     *@return: A string
     */
    static bool cmp(string str1, string str2) {
        return str1 + str2 > str2 + str1;
    }
    string int2str(int num) {
        string str = "";
        if (num == 0) return "0";
        stack<char> st;
        while (num != 0) {
            char ch = num % 10 + '0';
            num = num / 10;
            st.push(ch);
        }
        while (!st.empty()) {
            str += st.top();
            st.pop();
        }
        return str;
    }
    string largestNumber(vector<int> &num) {
        // write your code here
        vector<string> vec;
        for (int i = 0; i < num.size(); i ++) {
            vec.push_back(int2str(num[i]));
        }
        sort(vec.begin(), vec.end(), cmp);
        string str = "";
        int tag = 0;
        for (int i = 0; i < vec.size(); i ++) {
            if (vec[i] != "0") {
                tag = 1;
                str += vec[i];
            } else if (tag == 1 ) {
                str += vec[i];
            }
        }
        if (str == "") return "0";
        else return str;
    }
};
